src/megaclient.cpp

```diff
@@ -12907,20 +12817,26 @@ error MegaClient::isnodesyncable(Node *remotenode, bool *isinshare, SyncError *s
     bool inshare;
 
     // any active syncs below?
-    for (sync_list::iterator it = syncs.begin(); it != syncs.end(); it++)
-    {
-        if ((*it)->state == SYNC_ACTIVE || (*it)->state == SYNC_INITIALSCAN)
+    bool anyBelow = false;
+    syncs.forEachRunningSync([&](Sync* sync) {
+
+        if (sync->state == SYNC_ACTIVE || sync->state == SYNC_INITIALSCAN)
         {
-            n = (*it)->localroot->node;
+            n = sync->localroot->node;
 
             do {
                 if (n == remotenode)
                 {
-                    if(syncError) *syncError = ACTIVE_SYNC_BELOW_PATH;
-                    return API_EEXIST;
+                    anyBelow = true;
```

Comment:
We could break the loop and avoid going upwards in the tree, since we already found a sync below.
```suggestion
                    anyBelow = true;
                    break;
```

Reply:
I've resolved this one with a new function already added (based on review comments) in Pablo's branch - forEachRunningSyncContainingNode